testing/test_recwarn.py

```diff
@@ -37,6 +39,45 @@ def test_recwarn_captures_deprecation_warning(recwarn: WarningsRecorder) -> None
     assert recwarn.pop(DeprecationWarning)
 
 
+class TestSubclassWarningPop:
+    class ParentWarning(Warning):
+        pass
+
+    class ChildWarning(ParentWarning):
+        pass
+
+    class ChildOfChildWarning(ChildWarning):
+        pass
+
+    @staticmethod
+    def raise_warnings_from_list(_warnings: List[Type[Warning]]):
+        for warn in _warnings:
+            warnings.warn(f"Warning {warn().__repr__()}", warn)
+
+    def test_pop(self):
+        with pytest.warns((self.ParentWarning, self.ChildWarning)) as record:
+            self.raise_warnings_from_list(
+                [self.ChildWarning, self.ParentWarning, self.ChildOfChildWarning]
+            )
+
+        assert len(record) == 3
+        _warn = record.pop(self.ParentWarning)
+        assert _warn.category is self.ParentWarning
+
+    def test_pop_raises(self):
+        with pytest.raises(AssertionError):
+            with pytest.warns(self.ParentWarning) as record:
+                self.raise_warnings_from_list([self.ParentWarning])
+            record.pop(self.ChildOfChildWarning)
+
+    def test_pop_most_recent(self):
+        with pytest.warns(self.ParentWarning) as record:
+            self.raise_warnings_from_list([self.ChildWarning, self.ChildOfChildWarning])
```

Comment:
This test only happens to pass because (without the or-clause) we always match the first instance.

Use childofchild, child, childofchild.

Reply:
Oh yeah, it works now only with the or-clause